### ISSUE TITLE
Worker gets stuck mining to single pool

### ISSUE THREAD

**Issue by ZZCoder1** [RainbowMiner/RainbowMiner#1727]
My worker keeps getting stuck mining to one pool when there are other more profitable places to mine.
I have only seen it get stuck on Moneroocean. This is only happening on one rig, which is my server. Not sure if that is relevant.

It goes away sometimes, such as after I upgrade, but then starts again at some point.
Seem like a bug. Any ideas for a workaround?
 
[console edited.txt](https://github.com/RainbowMiner/RainbowMiner/files/7165043/console.edited.txt)

**Comment by RainbowMiner** [comment]
Could you please create and upload a Debug file for that machine? http://localhost:4000 - click "Debug file" on the left hand side. The created zip file has all sensitive data x-ed out, so that it's ready for upload.

**Comment by ZZCoder1** [comment]
It went away after I updated powershell and restarted RB, so that had some reset effect. 
Just restarting RB does not mitigate, and removing and adding moneroocean back to the pools also does not mitigate.
But this morning it is happening on a different worker, so it's not the server. Also stuck on moneroocean though.
Here is the debug from the new occurrence.
[ak_debug_2021-09-15.zip](https://github.com/RainbowMiner/RainbowMiner/files/7171359/ak_debug_2021-09-15.zip)

**Comment by ZZCoder1** [comment]
It happened again today. I found that pressing 'S' gets it unstuck and it resorts the algorithms in the correct order of profit. However, it still keeps mining for the min mining intervals before switching. As a side issue, I think 'S' should probably cancel the min mining requirement so it can switch right away.

**Comment by RainbowMiner** [comment]
That is so weird. Are these Windows or Linux machines?
You might try to switch to cURL: just create an empty text file "curl.txt" inside the RainbowMiner directory and restart RainbowMiner. Sometimes the Windows dynamic port stack gets depleted, resulting in rare crash conditions for the Powershell web request methods.

**Comment by RainbowMiner** [comment]
It would be great, if you could upload other debug files, in case that happens again. That makes it easier for me to track down the culprit.

**Comment by ZZCoder1** [comment]
These are Windows.

**Comment by ZZCoder1** [comment]
[debug_2021-09-26.zip](https://github.com/RainbowMiner/RainbowMiner/files/7233153/debug_2021-09-26.zip)

Looks like it happened again on yet another worker. Again, it is Moneroocean that is favored.

**Comment by ZZCoder1** [comment]
[debug_2021-09-27.zip](https://github.com/RainbowMiner/RainbowMiner/files/7238700/debug_2021-09-27.zip)

And again...

**Comment by RainbowMiner** [comment]
Could you please upload the most actual `errors_2021-09-27.asyncloader.txt` file, to be found in the Logs folder of that rig? 
I'll really want to find out, what is going on here.

**Comment by ZZCoder1** [comment]
That file contains my wallet addresses, so I'd rather not. If you add a support email address here or to your github profile I can send it there. It also is happening on another machine now, here is the new debug zip, in case it helps.
I can send a file from this machine as well, if you tell me which one and where to send it.

[ak_debug_2021-09-27.zip](https://github.com/RainbowMiner/RainbowMiner/files/7240251/ak_debug_2021-09-27.zip)
t.

**Comment by RainbowMiner** [comment]
Oh yes, you are right. Please send the file to info@rbminer.net

**Comment by ZZCoder1** [comment]
[ak_debug_2021-09-28.zip](https://github.com/RainbowMiner/RainbowMiner/files/7247962/ak_debug_2021-09-28.zip)
Here is another instance. All of the debug .zips contain errors_2021-xx-xx.asyncloader.txt except for debug_2021-09-27.zip. Not sure why.
Please confirm if you really need that particular instance of the error file. It is present in all 4 other .zip files.

**Comment by RainbowMiner** [comment]
> Please confirm if you really need that particular instance of the error file. It is present in all 4 other .zip files.

If the file is present, there is no need for that. I'll publish a new release today - the new version will try to encapsulate the web requests into separate jobs. This workaround might solve the asyncloader crash.

**Comment by ZZCoder1** [comment]
[ak_debug_2021-09-29.zip](https://github.com/RainbowMiner/RainbowMiner/files/7256328/ak_debug_2021-09-29.zip)
It is still happening on 4.7.4.9.

**Comment by RainbowMiner** [comment]
Oh dear. So encapsulating these calls into threadjobs doesn't help. Leaves us to try cURL instead of the Microsoft calls. I'll quickly add an option to the config, so that you can propagate this over your network.

**Comment by RainbowMiner** [comment]
Ok, I have added a new parameter to config.txt (release v4.7.5.0)
- add parameter "EnableCurl" to config.txt:
  - set to "1", if you get frequent "Asyncloader is crashed" warnings

Please wait until all server + clients have been updated to release v4.7.5.0, before setting that parameter.
Let me know, if the "Asyncloader is crashed" warning is gone when using cURL

**Comment by ZZCoder1** [comment]
[ak_debug_2021-10-04.zip](https://github.com/RainbowMiner/RainbowMiner/files/7279504/ak_debug_2021-10-04.zip)
Asyncloader crashed is not present, but it is still getting stuck mining to Moneroocean.

**Comment by RainbowMiner** [comment]
Thank you for reporting back! It is expected behaviour, since the database needs a bit, until all values (we use moving averages internally) are up to speed.

**Comment by ZZCoder1** [comment]
It happened again with EnableCurl set to 1. AsyncLoader is crashed. Is this supposed to implicitly disable AsyncLoader?

**Comment by RainbowMiner** [comment]
No, it doesn't disable the Asyncloader, but changes the web request function from Powershell to the external curl tool. This (should) avoid lock-ups during calls to the pool APIs. Could you please create another Debug file? Calls to curl include much more detailed logs and errors. Maybe I am able to furthermore isolate the problem, then.

**Comment by ZZCoder1** [comment]
[ak_debug_2021-10-04_2.zip](https://github.com/RainbowMiner/RainbowMiner/files/7287190/ak_debug_2021-10-04_2.zip)
